packages/integration-tests/src/e2e-flow/e2e_flow.test.ts

```diff
@@ -0,0 +1,440 @@
+import { execa } from 'execa';
+import * as fsp from 'fs/promises';
+import { existsSync } from 'fs';
+import * as path from 'path';
+import * as os from 'os';
+import { after, afterEach, before, beforeEach, describe, it } from 'node:test';
+import assert from 'assert';
+import { glob } from 'glob';
+import { BackendIdentifier } from '@aws-amplify/plugin-types';
+import {
+  CloudFormationClient,
+  DeleteStackCommand,
+} from '@aws-sdk/client-cloudformation';
+import { BackendIdentifierConversions } from '@aws-amplify/platform-core';
+import { getClientConfigPath } from '@aws-amplify/client-config';
+import { TestBranch, amplifyAppPool } from '../amplify_app_pool.js';
+import { testConcurrencyLevel } from '../test-e2e/test_concurrency.js';
+import { e2eToolingClientConfig } from '../e2e_tooling_client_config.js';
+
+type PackageManagerExecutable = 'npm' | 'yarn-classic' | 'yarn-modern' | 'pnpm';
+
+const concurrency = process.env.PACKAGE_MANAGER_EXECUTABLE?.startsWith('yarn')
+  ? 1
+  : testConcurrencyLevel;
+
+const packageManagerSetup = async (
+  packageManagerExecutable: PackageManagerExecutable,
+  dir?: string
+) => {
+  const execaOptions = {
+    cwd: dir || os.homedir(),
+    stdio: 'inherit' as const,
+  };
```

Comment:
Could we refactor the code in such a way that `process.env.PACKAGE_MANAGER_EXECUTABLE` is read only once somewhere?

Options:
1. Do that in `packageManagerSetup` and have `packageManagerSetup` return a type that provides all what test needs for further execution i.e. `type PackageManagerTestConfiguration = { executable: string; maxConcurrency?: number }`
2. Create `TestPackageManagerControllerFactory` and have `TestPackageManagerController` and subclasses (like we did in create amplify) to encapsulate functionalities.

(I think 1. is good enough for tests).

Also. If this stays as `packageManagerSetup` then it should be renamed to `setupPackageManager` and perhaps live in its own file given volume.